tools/monitoring/mysql/operations.go

```diff
@@ -67,3 +67,37 @@ func rollbackTx(tx *sql.Tx, err error) error {
 	}
 	return fmt.Errorf("Statement execution failed: %v; rollback failed: %v", err, rollbackErr)
 }
+
+// CheckAlertCondition checks whether the given error pattern meets
+// the alert condition specified in config
+func CheckAlertCondition(errorPattern string, config *config.SelectedConfig, db *sql.DB) (bool, error) {
+	// the timestamp we want to start collecting logs
+	startTime := time.Now().Add(time.Minute * time.Duration(config.Period))
+
+	_, err := db.Query(`
+		CREATE VIEW Matched AS
+  	SELECT Jobname, PrNumber 
+  	FROM ErrorLogs
+  	WHERE ErrorPattern=? and TimeStamp > ?`,
+		errorPattern, startTime)
+
+	if err != nil {
+		return false, err
+	}
+
+	var nMatches, nJobs, nPRs int
+
+	row := db.QueryRow(`
+		SELECT 
+    	COUNT(*),
+    	COUNT (DISTINCT Jobname),
+    	COUNT (DISTINCT PrNumber)
+		FROM Matched;`)
+
+	if err = row.Scan(&nMatches, nJobs, nPRs); err != nil {
```

Comment:
My suspicion is that this wouldn't work. Maybe you meant: 

```suggestion
	if err = row.Scan(&nMatches, &nJobs, &nPRs); err != nil {
```